File: backend/app/schemas/rule.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class RuleCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    description: str | None = None
    category: str | None = Field(default=None, max_length=100)
    condition: str = Field(min_length=1)
    action: str = Field(min_length=1)
    enabled: bool = True


class RuleUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = None
    category: str | None = Field(default=None, max_length=100)
    condition: str | None = Field(default=None, min_length=1)
    action: str | None = Field(default=None, min_length=1)
    enabled: bool | None = None


class RuleRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    description: str | None
    category: str | None
    condition: str
    action: str
    enabled: bool
    created_at: datetime

File: frontend/src/types/rule.ts
export interface WorkflowRule {
  id: string;
  name: string;
  description: string | null;
  category: string | null;
  condition: string;
  action: string;
  enabled: boolean;
  created_at: string;
}

export interface RuleCreatePayload {
  name: string;
  description?: string | null;
  category?: string | null;
  condition: string;
  action: string;
  enabled: boolean;
}

export interface RuleUpdatePayload {
  name?: string;
  description?: string | null;
  category?: string | null;
  condition?: string;
  action?: string;
  enabled?: boolean;
}
